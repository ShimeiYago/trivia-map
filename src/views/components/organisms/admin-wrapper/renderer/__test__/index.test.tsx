import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, Props } from '..';

let wrapper: ShallowWrapper<Props, unknown, Renderer>;

const basicProps: Props = {
  user: {
    userId: 1,
    nickname: 'Axel',
    email: 'xxx@example.com',
    icon: 'https://...',
    url: 'https://...',
    isSocialAccount: false,
  },
  autoLoggingInState: 'success',
  loggedOutSuccessfully: false,
  isMobile: false,
  children: <div />,
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<Renderer {...basicProps} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mobile', () => {
    wrapper.setProps({ isMobile: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('loading', () => {
    wrapper.setProps({ autoLoggingInState: 'loading' });
    expect(wrapper).toMatchSnapshot();
  });

  it('redirect to login page', () => {
    wrapper.setProps({ autoLoggingInState: 'error' });
    expect(wrapper).toMatchSnapshot();
  });

  it('redirect to map page when logged out', () => {
    wrapper.setProps({ autoLoggingInState: 'error', loggedOutSuccessfully: true });
    expect(wrapper).toMatchSnapshot();
  });
});
