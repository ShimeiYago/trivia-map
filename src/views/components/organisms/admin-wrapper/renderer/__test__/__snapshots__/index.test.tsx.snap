// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<SingleRowPageWrapper>
  <div />
</SingleRowPageWrapper>
`;

exports[`Shallow Snapshot Tests loading 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "mt": 3,
    }
  }
>
  <CenterSpinner />
</ForwardRef(Box)>
`;

exports[`Shallow Snapshot Tests redirect to login page 1`] = `
<Navigate
  to="/login"
/>
`;

exports[`Shallow Snapshot Tests redirect to map page when logged out 1`] = `
<Navigate
  to="/"
/>
`;
